web_sdk/BUILD.gn

```diff
@@ -268,7 +288,8 @@ template("_compile_platform") {
   assert(defined(invoker.output_dill),
          "output_dill must be defined for $target_name")
   _kernel_worker(target_name) {
-    inputs = [ "sdk_rewriter.dart" ] + web_ui_sources
+    inputs =
+        [ "sdk_rewriter.dart" ] + web_ui_sources + web_locale_keymap_sources
```

Comment:
I'm doing a similar change in [another PR](https://github.com/flutter/engine/pull/37440) and I don't think this line is necessary.

Reply:
I still think this is needed, because `web_ui_sources` only contains all files within `//flutter/lib/web_ui/lib`, while `web_locale_keymap_sources` imports all files within `//flutter/third_party/web_locale_keymap/lib`.

Reply:
Actually after reading through this, I think in general the values for `inputs`, both in `_kernel_worker` and in `_dartdevc` are actually sort of bogus. Both of them operate on the already rewritten sources, but are using the pre-rewrite sources as inputs. This kind of works, just because it's sort of transitively capturing the chain of causality. That is to say, the target knows that if `sdk_rewriter.dart` or any of the input files change, the script will be rerun and the outputs of the script might possibly change.

This is kind of janky, however, and what would actually be better is for the `sdk_rewriter` template to accurately reflect each output file that it creates, and have the inputs of the `_kernel_worker` or `_dartdevc` templates be the output of the `sdk_rewriter` template. That's probably out of scope for this change though. I think for now, having `web_locale_keymap_sources` in the inputs is actually probably slightly more correct than not having them. That way, if something changes in `web_locale_keymap_sources` it causes us to rebuild the kernels and JS/sourcemaps.

Reply:
I went ahead and filed a follow-on issue here: https://github.com/flutter/flutter/issues/116195

If you are feeling adventurous and want to fix the build rules as part of this PR, have at it. Otherwise, would you mind putting a `TODO(jacksongardner)` with a link to this github issue below it where we are specifying these inputs?